src/App.js

```diff
@@ -1,16 +1,72 @@
-import React from 'react';
+import React, { Component } from 'react'
+import './components/TodoComponents/Todo.css'
+import TodoList from './components/TodoComponents/TodoList'
+import TodoForm from './components/TodoComponents/TodoForm'
 
-class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+class App extends Component {
+  inputElement = React.createRef()
+  constructor() {
+    super()
+    this.state = {
+      items: [],
+      currentItem: {
+        text: '',
+        key: '',
+        completed:false
+      },
+    }
+  }
+  deleteItem = key => {
+    var filteredItems = this.state.items.filter(item => {
+      return item.key === key
```

Comment:
I wouldn't have thought to use `.filter()` here, but it totally works. How would you do it with `.map()`

What if someone completed a task by accident? How could you make it so that, after a click, completed would be set to false if the Todo had already been completed?